impl/pkg/service/pkarr.go

```diff
@@ -146,11 +169,13 @@ func fromPkarrRecord(record storage.PkarrRecord) (*GetPkarrResponse, error) {
 // GetPkarr returns the full Pkarr record (including sig data) for the given z-base-32 encoded ID
 func (s *PkarrService) GetPkarr(ctx context.Context, id string) (*GetPkarrResponse, error) {
 	// first do a cache lookup
-	if s.cache.Has(id) {
-		cacheItem := s.cache.Get(id)
-		until := time.Until(cacheItem.ExpiresAt())
-		logrus.Debugf("resolved pkarr record<%s> from cache, with remaining TTL: %s", id, until.String())
-		return fromPkarrRecord(cacheItem.Value())
+	if got, err := s.cache.Get(id); err == nil {
```

Comment:
Curious, why not cache the exact same thing that will be returned? The benefit is that less serialization would happen during the read path, which is presumably the one that will get lots of load.